pkg/providers/cloudstack/validator.go

```diff
@@ -28,104 +30,148 @@ var restrictedUserCustomDetails = [...]string{
 	"keypairnames", "controlNodeLoginUser",
 }
 
-var domainId string
-
-func NewValidator(cmk ProviderCmkClient) *Validator {
+func NewValidator(cmks CmkClientMap) *Validator {
 	return &Validator{
-		cmk: cmk,
+		cmks:                   cmks,
+		localAvailabilityZones: []localAvailabilityZone{},
```

Comment:
I don't think you should have `localAvailabilityZones` in the validator struct if you are generating it from one of the validation method.

Reply:
Good point